src/draw/opengles/lv_draw_opengles.c

```diff
@@ -0,0 +1,666 @@
+            }
+            break;
+        case LV_DRAW_TASK_TYPE_LINE: {
+                lv_draw_line_dsc_t line_dsc;
+                lv_memcpy(&line_dsc, task->draw_dsc, sizeof(line_dsc));
+                line_dsc.base.user_data = (void *)(uintptr_t)1;
+                lv_draw_line(&dest_layer, &line_dsc);
+            }
+            break;
+        case LV_DRAW_TASK_TYPE_TRIANGLE: {
+                lv_draw_triangle_dsc_t triangle_dsc;
+                lv_memcpy(&triangle_dsc, task->draw_dsc, sizeof(triangle_dsc));
+                triangle_dsc.base.user_data = (void *)(uintptr_t)1;
+                lv_draw_triangle(&dest_layer, &triangle_dsc);
+            }
+            break;
+        case LV_DRAW_TASK_TYPE_IMAGE: {
+                lv_draw_image_dsc_t image_dsc;
+                lv_memcpy(&image_dsc, task->draw_dsc, sizeof(image_dsc));
+                image_dsc.base.user_data = (void *)(uintptr_t)1;
+                lv_draw_image(&dest_layer, &image_dsc, &task->area);
+                break;
+            }
+        default:
+            return false;
+    }
+
+    while(dest_layer.draw_task_head) {
+        lv_draw_dispatch_layer(disp, &dest_layer);
+        if(dest_layer.draw_task_head) {
+            lv_draw_dispatch_wait_for_request();
+        }
+    }
+
+    // unsigned int texture = SDL_CreateTexture(lv_sdl_window_get_renderer(disp), SDL_PIXELFORMAT_ARGB8888,
+    //                                           SDL_TEXTUREACCESS_STATIC, texture_w, texture_h);
+    // SDL_SetTextureBlendMode(texture, SDL_BLENDMODE_BLEND);
+    // SDL_UpdateTexture(texture, NULL, opengles_render_buf, texture_w * 4);
+
+    unsigned int texture;
+
+    GL_CALL(glGenTextures(1, &texture));
+    GL_CALL(glBindTexture(GL_TEXTURE_2D, texture));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE));
+    GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE));
+    GL_CALL(glPixelStorei(GL_UNPACK_ALIGNMENT, 1));
+
+    /*Color depth: 8 (L8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888)*/
+#if LV_COLOR_DEPTH == 8
+    GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, texture_w, texture_h, 0, GL_RED, GL_UNSIGNED_BYTE, opengles_render_buf));
+#elif LV_COLOR_DEPTH == 16
+    GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB565, texture_w, texture_h, 0, GL_RGB, GL_UNSIGNED_SHORT_5_6_5,
+                         opengles_render_buf));
+#elif LV_COLOR_DEPTH == 24
+    GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, texture_w, texture_h, 0, GL_BGR, GL_UNSIGNED_BYTE, opengles_render_buf));
+#elif LV_COLOR_DEPTH == 32
+    GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, texture_w, texture_h, 0, GL_BGRA, GL_UNSIGNED_BYTE,
+                         opengles_render_buf));
+#else
+#error("Unsupported color format")
+#endif
+
+    glGenerateMipmap(GL_TEXTURE_2D);
```

Comment:
I usually follow this call with:
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL _mipMapLevel)
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR)
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR)
Or nearest when chosing not to use interpolation.

The first call to lower the default index value of 1000 mipmaplevels to a value more in the range of 1-12. 
The latter might apply more to 3d space where GL_LINEAR_MIPMAP_LINEAR gives the smoothest transition between mipmaps but I use it for 2D purposes as well.

Reply:
Thanks! 1000 is a lot of mipmap levels, especially considering the mipmap is unused since the textures are currently never being scaled relative to the pixels. I had been wondering if we could avoid generating the mipmap. I recall trying to remove it and something broke.

We could probably accelerate LVGL object/image transformations with OpenGL. That can be a future PR. I imagine the mipmap will be important to have for that. In the meantime I'll apply your suggestion to be well-prepared for forward compatibility.

Reply:
I tried linear but that gave some bleeding between labels and background due to interpolation. You want this in 3D space to resolve some artifacts in depth but it doesn't look as good in 2D. Setting To Nearest is better in this case.
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 20);
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST_MIPMAP_NEAREST);
glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);